### PR TITLE
[MLIR][TORCH] Add e2e support `aten.movedim.int` op

### PR DESCRIPTION
**Pull request by vivekkhandelwal1**
Signed-Off By: Vivek Khandelwal<vivek@nod-labs.com>

### CODE REVIEW

File: python/torch_mlir/dialects/torch/importer/jit_ir/build_tools/abstract_interp_lib_gen.py [hunk 1]
```diff
@@ -370,6 +370,45 @@ def aten〇sum〇dim_IntList〡shape(self: List[int], dim: Optional[List[int]],
 def aten〇permute〡shape(self: List[int], dims: List[int]) -> List[int]:
     return upstream_shape_functions.permute(self, dims)
 
+def _movedim(self: List[int], source: List[int], destination: List[int]) -> List[int]:
```
Comment (ramiro050) [inline python/torch_mlir/dialects/torch/importer/jit_ir/build_tools/abstract_interp_lib_gen.py:373]: Can you upstream this first?

**Review by ramiro050 — CHANGES_REQUESTED** [review]

File: lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp [hunk 2]
```diff
@@ -3402,6 +3402,75 @@ class DecomposeAtenVarMeanOp : public OpRewritePattern<AtenVarMeanOp> {
...
+      SmallVector<int64_t> sourceDims, destinationDims;
+      for (unsigned i = 0; i < inputRank; i++) {
+        if (srcDims[i] != -1)
+          sourceDims.push_back(srcDims[i]);
+        if (dstDims[i] != -1)
+          destinationDims.push_back(dstDims[i]);
+      }
+      unsigned restDim = inputRank - 1;
+      for (unsigned i = 0; i < restDim; i++)
+        dimsOrder[destinationDims[i]] = sourceDims[i];
```
Comment (ramiro050) [inline lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp:3456]: I think you can do instead

```c++
  SmallVector<int> dimsOrder(llvm::iota_range<int>(0, inputRank, /*inclusive=*/false));
  dimsOrder.erase(dimsOrder.begin() + srcDimInt);
  dimsOrder.insert(dimsOrder.begin() + dstDimInt, srcDimInt);
```

File: python/torch_mlir_e2e_test/test_suite/basic.py [hunk 3]
```diff
@@ -3123,4 +3123,42 @@ def forward(self):
...
+
+class MoveDimIntModule(torch.nn.Module):
+
+    def __init__(self):
+        super().__init__()
+
+    @export
+    @annotate_args([None, ([3, 4, 2, 1], torch.float32, True)])
+    def forward(self, x):
+        return torch.movedim(x, source=1, destination=2) #0, 2, 1
```
Comment (ramiro050) [inline python/torch_mlir_e2e_test/test_suite/basic.py:3140]: nit: use `torch.ops.aten.movedim`, same below. See:

https://github.com/llvm/torch-mlir/blob/main/docs/adding_an_e2e_test.md#forward-method

**Review by ramiro050 — CHANGES_REQUESTED** [review]

File: lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp (hunk 2 above)
Comment (vivekkhandelwal1) [reply lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp:3456]: The algo is taken from here: https://github.com/pytorch/pytorch/blob/c47bdd7522c7146c091c95ff5b4ce354d7a8870c/aten/src/ATen/native/TensorShape.cpp#L3660. I don't understand how these 3 lines of code would work in place of this algo.

**Comment by vivekkhandelwal1** [comment]
I'm closing this PR, since this patch has been moved to this PR: https://github.com/llvm/torch-mlir/pull/1773

**Pull request closed without merging**